src/integration.rs

```diff
@@ -303,7 +303,7 @@ macro_rules! shared_event_type {
             pub fn set_details<T: ?Sized>(mut self, details: &T) -> Self
                 where T: Serialize
             {
-                self.details = Some(to_value(details));
+                self.details = Some(to_value(details).unwrap());
```

Comment:
Not happy about these two unwraps, but given they're already present in the other parts of this file, I've left it as is for now to avoid changing other parts of the api.

Reply:
This seems fine, but just so nobody is surprised when it panics, could you please add a doc comment with a `# Panics` section that explains if your serialization is fallible and it fails, this will panic.